src/hash_generator/EPCISEventToPreHashedString.js

```diff
@@ -0,0 +1,449 @@
+      for (let i = 0; i < list.length; i += 1) {
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          quantityElementCanonicalPropertyOrder, throwError);
+        strings.push(`quantityElement${res.preHashed}`);
+        customFields.push(...res.customFields.map((j) => `quantityElement${j}`));
+      }
+      break;
+    case 'bizTransactionList':
+      for (let i = 0; i < list.length; i += 1) {
+        // if, for example, the field is equal to 'bol' instead of
+        // 'urn:epcglobal:cbv:btt:bol', we need to complete it
+        if (businessTransactionTypes[list[i].type] !== undefined) {
+          list[i].type = businessTransactionTypes[list[i].type];
+        }
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          bizTransactionCanonicalPropertyOrder, throwError);
+        strings.push(res.preHashed);
+        customFields.push(...res.customFields.map((j) => `bizTransaction${j}`));
+      }
+      break;
+    case 'sourceList':
+      for (let i = 0; i < list.length; i += 1) {
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          sourceCanonicalPropertyOrder, throwError);
+        strings.push(res.preHashed);
+        customFields.push(...res.customFields.map((j) => `source${j}`));
+      }
+      break;
+    case 'destinationList':
+      for (let i = 0; i < list.length; i += 1) {
+        // if, for example, the field is equal to 'location' instead of
+        // 'urn:epcglobal:cbv:sdt:location', we need to complete it
+        if (sourceDestinationTypes[list[i].type] !== undefined) {
+          list[i].type = sourceDestinationTypes[list[i].type];
+        }
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          destinationCanonicalPropertyOrder, throwError);
+        strings.push(res.preHashed);
+        customFields.push(...res.customFields.map((j) => `destination${j}`));
+      }
+      break;
+    case 'sensorElementList':
+      for (let i = 0; i < list.length; i += 1) {
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          sensorElementCanonicalPropertyOrder, throwError);
+        strings.push(`sensorElement${res.preHashed}`);
+        customFields.push(...res.customFields.map((j) => `sensorElement${j}`));
+      }
+      break;
+    case 'sensorReport':
+      string = '';
+      for (let i = 0; i < list.length; i += 1) {
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
+          sensorReportCanonicalPropertyOrder, throwError);
+        strings.push(`sensorReport${res.preHashed}`);
+        customFields.push(...res.customFields.map((j) => `sensorReport${j}`));
+      }
+      break;
+    case 'correctiveEventIDs':
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('correctiveEventID', list[i], throwError));
+      }
+      break;
+    // no default
```

Comment:
What happens in this case? All other fields are ignored? Should it warn?

Reply:
You are right. We need a default case if there is a custom field at the root of an event that is an array